### ISSUE TITLE
[expo-camera] [web] Errors on older browsers due to using "new" MediaStream APIs

### ISSUE THREAD

**Issue by NiciusB** [expo/expo#6194]
### Environment

Motorola G4
Chrome Mobile, Version: 55.0.2883
Android, Version: 7.0

This is an extremely old Chrome version. About 2% of our users would be affected by this.

### Steps to Reproduce

Mount a \<Camera\/\> component

### Expected Behavior

The camera should mount and have basic capabilities (Like taking a picture), even if not all of the MediaStream APIs are available

### Actual Behavior

It throws an error due to `this.settings = this.stream.getTracks()[0].getSettings()`
This could be avoided by not setting the settings on older devices, for example:
```
const tracks = this.stream.getTracks()
if (tracks.length && tracks[0].getSettings) {
  this.settings = tracks[0].getSettings()
}
```

After changing that code, it mounts correctly, but shortly after it throws another error due to `track.getCapabilities()` on `onCapabilitiesReady`
A quick way to avoid this is to check for the API being available at the start of the function
```
async onCapabilitiesReady(track) {
  if (!track.getCapabilities) return
```

### Reproducible Demo

You'll need a device running an older version of Chrome Mobile (<v61 for the getSettings error, <v66 for the getCapabilities error, as listed on the MDN web docs)

Website: https://balbona.me/expowebcamera-example/web-build/
Code:
```
import * as React from "react";
import { Camera } from "expo-camera";

export default class App extends React.Component {
  render() {
    return (
      <Camera
        style={{ flex: 1, backgroundColor: "#90f1f1" }}
        onMountError={({ message }) => alert("onMountError: " + message)}
      />
    );
  }
}

```

### Other concerns

I have not dug up into the code too much, so I'm not sure what the effects of the changes suggested above could have. Afaik it works fine, and I'm able to take photos without issues.

Expo might not want to support those browsers (Chrome Mobile 66 was launched in 2018/04/17). If that's the case at least this is documented now!

**Comment by EvanBacon** [comment]
Thank you for your thorough report. We should at least implement a `isAvailableAsync` feature for legacy users.

**Comment by NiciusB** [comment]
`isAvailableAsync` would be great, both for older devices and devices without a camera.

I think my code change suggestions are also interesting, since they would increase availability. Again, I'm not sure of the consequences of these but they didn't seem to negatively affect anything when I tested it on Chrome Mobile 55

**Comment by EvanBacon** [comment]
Fixed in #6207
